src/support/utf8.c

```diff
@@ -615,71 +615,63 @@ size_t u8_printf(const char *fmt, ...)
     return cnt;
 }
 
-/* based on the valid_utf8 routine from the PCRE library by Philip Hazel
+/* Rewritten completely, original code not based on anything else
 
    length is in bytes, since without knowing whether the string is valid
    it's hard to know how many characters there are! */
-int u8_isvalid(const char *str, size_t length)
+int u8_isvalid(const char *iStr, size_t iLength)
```

Comment:
@StefanKarpinski Because when I had copied code of mine previously into `utf8.c`, I was told to remove the documentation, because nobody had decided yet how code should be documented :cry:

@tknopp I don't mean to sound snide, but, if I'd actually found any consistent conventions in the code I've been looking at, I would have followed it...
I've been working mostly in `utf8.c`, `utf8proc.c`, and `gc.c`.  Each one has different conventions for how the code is formatted, [2 vs 4 spaces for indentation, for example, different ways of indenting {}'s, etc.], and different ways things are named. There are things like `jlegal_func` (is that a legal function? hmmm...) or should that really have been `jl_egal_func`, as were some of the other functions, such as `jl_alloc_svec_func`.

Having authored the coding / style / portability standard for a company with hundreds of programmers, that is still in use after 28 years, and co-authored the guide for the start-up I'm at now, I feel not having some sort of guide for C/C++ programming is a serious lack for Julia.
I've found that consistency, external and internal documentation, good readable style, as well as having portable code, is incredibly important when you have lots of contributors over large periods of time.
The code now is not very consistent, there doesn't seem to be any internal documentation or comments to explain whatever tricks or design / implementation decisions were made, use of too short variable names makes it hard to read parts of the code (using a single letter variable name, that is a substring of several other variable names in the same function, for example)...
lots of static variables that will make adding threading very difficult... I could go on, but it would just frustrate me more...